File: examples/core/target-lines/src/main/python/TargetLines.py
import sys
from pyspark import SparkContext

if __name__ == "__main__":

  if len(sys.argv) != 3:
    print >> sys.stderr, "Usage: TargetLines <path> <list>"
    exit(-1)

  sc = SparkContext()
  listbc = sc.broadcast(sys.argv[2].split(" "))
  targetLines = sc.textFile(sys.argv[1]).filter(lambda line: any(v in line for v in listbc.value))
  for line in targetLines.collect(): print(line)
  sc.stop()

File: examples/core/target-lines/src/main/scala/my/spark/core/TargetLines.scala
package my.spark.core

import org.apache.spark.SparkContext
import org.apache.commons.lang.StringUtils

object TargetLines {

  def main(args: Array[String]) {

    if (args.length != 2) {
      System.err.println("Usage: TargetLines <path> <list>")
      System.exit(1)
    }

    val sc = new SparkContext()
    val listbc = sc.broadcast(args(1).split(" "))
    val targetLines = sc.textFile(args(0)).filter(line => listbc.value.count(StringUtils.containsIgnoreCase(line,_)) > 0)
    targetLines.foreach(println)
    sc.stop()
  }
}
